Source/EasyNetQ.Management.Client/ManagementClient.cs

```diff
@@ -592,71 +605,72 @@ public Task DeletePermissionAsync(Permission permission, CancellationToken cance
     {
         Ensure.ArgumentNotNull(permission, nameof(permission));
 
-        return DeleteAsync($"permissions/{permission.Vhost}/{permission.User}", cancellationToken);
+        return DeleteAsync(Permissions / permission.Vhost / permission.User, cancellationToken);
     }
 
     public Task<IReadOnlyList<TopicPermission>> GetTopicPermissionsAsync(CancellationToken cancellationToken = default)
     {
-        return GetAsync<IReadOnlyList<TopicPermission>>("topic-permissions", cancellationToken);
+        return GetAsync<IReadOnlyList<TopicPermission>>(TopicPermissions, cancellationToken);
     }
 
     public Task CreateTopicPermissionAsync(TopicPermissionInfo topicPermissionInfo, CancellationToken cancellationToken = default)
     {
         Ensure.ArgumentNotNull(topicPermissionInfo, nameof(topicPermissionInfo));
 
         return PutAsync(
-            $"topic-permissions/{SanitiseVhostName(topicPermissionInfo.GetVirtualHostName())}/{topicPermissionInfo.GetUserName()}",
-            topicPermissionInfo, cancellationToken
+            TopicPermissions / topicPermissionInfo.GetVirtualHostName() / topicPermissionInfo.GetUserName(),
+            topicPermissionInfo,
+            cancellationToken
         );
     }
 
     public Task DeleteTopicPermissionAsync(TopicPermission topicPermission, CancellationToken cancellationToken = default)
     {
         Ensure.ArgumentNotNull(topicPermission, nameof(topicPermission));
 
-        return DeleteAsync($"topic-permissions/{topicPermission.Vhost}/{topicPermission.User}", cancellationToken);
+        return DeleteAsync(TopicPermissions / topicPermission.Vhost / topicPermission.User, cancellationToken);
     }
 
     public Task<List<Federation>> GetFederationAsync(CancellationToken cancellationToken = default)
     {
-        return GetAsync<List<Federation>>("federation-links", cancellationToken);
+        return GetAsync<List<Federation>>(FederationLinks, cancellationToken);
     }
 
     public async Task<bool> IsAliveAsync(Vhost vhost, CancellationToken cancellationToken = default)
     {
         Ensure.ArgumentNotNull(vhost, nameof(vhost));
 
         var result = await GetAsync<AlivenessTestResult>(
-            $"aliveness-test/{SanitiseVhostName(vhost.Name)}", cancellationToken
+            AlivenessTest / vhost.Name, cancellationToken
         ).ConfigureAwait(false);
         return result.Status == "ok";
     }
 
-    private Task<T> GetAsync<T>(string path, CancellationToken cancellationToken = default)
+    private Task<T> GetAsync<T>(in RelativePath path, CancellationToken cancellationToken = default)
     {
         return GetAsync<T>(path, null, cancellationToken);
     }
 
 
     private async Task<T> GetAsync<T>(
-        string path,
+        RelativePath path,
```

Comment:
In?

Reply:
I see it on line 649 but not in other methods.

Reply:
It is not supported in async methods. Probably I should remove it here.